### PR TITLE
Incremental PR towards #347

### PR DESCRIPTION
**Pull request by kolharsam**
This PR is an incremental PR against #347 based on this [comment](https://github.com/s-yadav/react-number-format/pull/347#issuecomment-683865867)

### CODE REVIEW

**Comment by kolharsam** [comment]
@s-yadav I'm not sure if a changelog entry is needed for this PR. I could just add one for #347 directly.

**Comment by s-yadav** [comment]
@kolharsam Can you add failing test case, which this PR fixes.

**Comment by kolharsam** [comment]
@s-yadav I have added some tests with regards to these changes but I failed to think of a failing test. Would passing an empty string be considered a failing test? Would appreciate some input here. If so, then we could change the condition to just be `!props.value ? .... : .....` [here](https://github.com/s-yadav/react-number-format/pull/462/files#diff-a1be15e225934962f6ce9ddb7475f5945550187846ed4cd69c3d0a51f849abbcR152)

**Review by s-yadav — APPROVED** [review]

**Comment by s-yadav** [comment]
Looks good. Approved.

**Comment by s-yadav** [comment]
@kolharsam Looks like some specs are failing after this merge. Can you check that.

**Comment by kolharsam** [comment]
@s-yadav I think the specs are failing because the original PR #347 hasn't been merged yet and these changes were increments over the mentioned PR. So yeah, I guess things would get fixed if we merge #347.

**Comment by s-yadav** [comment]
Merged. Working.

File: test/library/input.spec.js [hunk 1]
```diff
@@ -56,6 +56,33 @@ describe('NumberFormat as input', () => {
     expect(wrapper.find('input').instance().value).toEqual('$2,456,981');
   });
 
+  it('should load the default value when initial value is null', () => {
+    const wrapper = mount(<NumberFormat value={null} defaultValue={89} />);
+    expect(wrapper.state().value).toEqual('89');
```
Comment (s-yadav) [inline test/library/input.spec.js:61]: We should never check for state value. As it's implementation details. Instead we should test for the end side-effect.
```js
expect(wrapper.find('input').instance().value).toEqual('89');
```

File: test/library/input.spec.js [hunk 2]
```diff
@@ -56,6 +56,33 @@ describe('NumberFormat as input', () => {
...
+  });
+
+  it('should load the prevous valid value if the state is changed to null', () => {
```
Comment (s-yadav) [inline test/library/input.spec.js:64]: Typo on the statement.
```
  should load the previous valid value
```

File: test/library/input.spec.js [hunk 3]
```diff
@@ -56,6 +56,33 @@ describe('NumberFormat as input', () => {
...
+        return (<NumberFormat value={this.state.testState} />)
+      }
+    }
+
+    const wrapper = mount(<WrapperComponent />);
+    const input = wrapper.find('input');
+    const domInput = input.instance();
+
+    expect(domInput.value).toEqual('90');
+    wrapper.setState({testState: null});
```
Comment (s-yadav) [inline test/library/input.spec.js:82]: This spec could be simplified you just mount `<NumberFormat value={this.state.testState} />` and then change the props through setProps.

Also FYI. Try avoiding interacting with states in specs. You shouldn't depend much on implementation detail.

**Comment by s-yadav** [comment]
@kolharsam The test is still failing when I am trying to run the test.
Screenshot 2020-12-18 at 12 18 07 PM

Also, missed reviewing specs earlier. Added some comments there.

**Comment by s-yadav** [comment]
@kolharsam Can you try pulling code from origin and rerun the test.

**Comment by kolharsam** [comment]
Will be making a new PR to fix this ASAP

**Pull request merged**